### ISSUE TITLE
not_unsafe_ptr_arg_deref does not trigger on type aliases and trait methods

### ISSUE THREAD

**Issue by SeeSpring** [rust-lang/rust-clippy#8092]
### Summary

If the raw pointer is used behind a `type` alias or the function is part of a trait, not_unsafe_ptr_arg_deref is not triggered.

### Lint Name

not_unsafe_ptr_arg_deref

### Reproducer

I tried this code:

```rust
trait Bad {
    fn f(x: *const u8) -> u8 {
        unsafe { *x }
    }
}

type Alias = *const u8;
pub fn bad(x: Alias) -> u8 {
    unsafe { *x }
}

pub fn main() {
    bad(0 as _);
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c52d2035d080cf0a95ae50757aa0c62e)

I expected to see this happen: Lint for `not_unsafe_ptr_arg_deref` is triggered

Instead, this happened: Lint is not triggered

**Comment by asquared31415** [comment]
This was fixed in #8273
